tests/beaker_tests/cisco_vrf/test_vrf.rb

```diff
@@ -108,104 +43,47 @@
   },
 }
 
-tests['non_default_properties_misc'] = {
+# Non-default Tests. NOTE: [:resource] = [:manifest_props] for all non-default
+tests[:non_default] = {
   desc:           '2.1 Non Default Properties commands',
   title_pattern:  'blue',
   manifest_props: {
-    description: 'test desc',
-    shutdown:    'true',
-  },
-  resource:       {
-    'description' => 'test desc',
-    'shutdown'    => 'true',
-  },
-}
-
-tests['non_default_properties_rd'] = {
-  desc:           '2.2 Non Default Properties RD',
-  title_pattern:  'blue',
-  platform:       'n(7|9)k',
-  manifest_props: {
-    route_distinguisher: '1:1'
-  },
-  resource:       {
-    'route_distinguisher' => '1:1'
-  },
-}
-
-tests['non_default_properties_vni'] = {
-  desc:           '2.3 Non Default Properties VNI',
-  title_pattern:  'blue',
-  platform:       'n9k',
-  manifest_props: {
-    vni: '4096'
-  },
-  resource:       {
-    'vni' => '4096'
+    description:         'test desc',
+    route_distinguisher: '1:1',
+    shutdown:            'true',
+    vni:                 '4096',
   },
 }
 
-# Create actual manifest for a given test scenario.
-def build_manifest_vrf(tests, id)
-  manifest = prop_hash_to_manifest(tests[id][:manifest_props])
-  if tests[id][:ensure] == :absent
-    state = 'ensure => absent,'
-    tests[id][:resource] = { 'ensure' => 'absent' }
-  else
-    state = 'ensure => present,'
-  end
-
-  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
-  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
-  \nnode default {
-  cisco_vrf { '#{tests[id][:title_pattern]}':
-    #{state}\n#{manifest}
-  }\n}\nEOF"
-
-  cmd = PUPPET_BINPATH + "resource cisco_vrf '#{tests[id][:title_pattern]}'"
-  tests[id][:resource_cmd] = get_namespace_cmd(agent, cmd, options)
-end
-
-# Wrapper for vrf specific settings prior to calling the
-# common test_harness.
-def test_harness_vrf(tests, id)
-  return unless platform_supports_test(tests, id)
-
-  tests[id][:ensure] = :present if tests[id][:ensure].nil?
-
-  # Workaround for (ioctl) facter bug on n7k ***
-  tests[id][:code] = [0, 2] if platform[/n7k/]
-
-  # Build the manifest for this test
-  build_manifest_vrf(tests, id)
-
-  test_harness_common(tests, id)
-  tests[id][:ensure] = nil
+def unsupported_properties(_tests, _id)
+  return [] if operating_system == 'nexus'
```

Comment:
In the old code above, it looks like route_distinguisher is unsupported except on n7k/n9k and vni is unsupported except on n9k. Am I missing where that check is now enforced?